src/runner.py

```diff
@@ -716,7 +716,9 @@ def _configure_docker_proxy(self) -> None:
                 }
             }
         }
-        docker_client_proxy_content = json.dumps(docker_client_proxy)
+        docker_client_proxy_content = json.dumps(
+            docker_client_proxy, default=lambda x: x.unicode_string()
```

Comment:
What is the reasoning here please?

Reply:
`docker_client_proxy` is a Url class and Url classes can not be serialized. To fix that we turn them into strings then do json.dump. The reason we use default and not just directly str cast is it can be None.